### PR TITLE
Setup error handling component and state

### PR DESCRIPTION
**Pull request by janitastic**
:bangbang: **_Please don't merge until we review and chat about this together_** :bangbang:

Is this a fix or a feature?
- [ ] Bug fix
- [x] Feature
- [ ] Refactor
 
Describe the feature - fix - refactor:
- Error handling component that is capturing a status code. 
- This is in state. Would love to chat about moving it to App so we can pass it through to each component.

How was this tested?
- [ ] open index.html
- [x] localhost
- [ ] terminal

Where should the reviewer start - test:
- NetworkError component. 

Additional comments:

### CODE REVIEW

**Comment by devonwigle** [comment]
Did you get a chance to chat about it?

File: src/Components/ConfirmationMessages/DonorThankYou.js [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
-  return (
-    <article className='main-container message'>
-      <h2 data-testid='thank-donor-title'>Thank You!</h2>
-      <p>Thank you for offering to donate milk. An eager parent will be in contact with you soon!</p>
-    </article>
-  )
+  const { error, loading, data } = useQuery(GET_ALL_DONORS);
+  const [errorStatus, setError] = useState(false);
+
+  useEffect(() => {
```
Comment (cagallo) [inline src/Components/ConfirmationMessages/DonorThankYou.js:9]: genius with the useEffect!!

File: src/Components/ConfirmationMessages/DonorThankYou.js [hunk 2]
```diff
@@ -1,10 +1,34 @@
...
+    if(!data && loading) {
+      setError(true)
+    }
+  }, [data])
+
+  if (errorStatus) {
+    console.log('errorStatus after setError', errorStatus)
```
Comment (cagallo) [inline src/Components/ConfirmationMessages/DonorThankYou.js:16]: hahaha, we will clean these out later!

**Comment by janitastic** [comment]
Yeah! Chez and I met with my mentor this morning and resolved a few things. Everything is working now, yay!

File: src/Components/NotFound/NotFound.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import './NotFound.scss';
+
+const NotFound = () => {
```
Comment (cagallo) [inline src/Components/NotFound/NotFound.js:4]: I love this component!

**Review by cagallo — APPROVED** [review]
Wow, incredible work, most challenging part of the project!

**Pull request merged**